pkg/apis/autoscaling/validation/validation.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	apivalidation "k8s.io/kubernetes/pkg/api/validation"
+	"k8s.io/kubernetes/pkg/apis/autoscaling"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+// ValidateHorizontalPodAutoscaler can be used to check whether the given autoscaler name is valid.
+// Prefix indicates this name will be used as part of generation, in which case trailing dashes are allowed.
+func ValidateHorizontalPodAutoscalerName(name string, prefix bool) (bool, string) {
+	// TODO: finally move it to pkg/api/validation and use nameIsDNSSubdomain function
```

Comment:
I think you can directly use NameIsDNSSubdomain() method? We do that for daemon sets and deployment: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/extensions/validation/validation.go#L236